lib/features/notes/data/repositories/notifications_repository.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:dairy_app/core/logger/logger.dart';
+import 'package:dairy_app/features/notes/domain/repositories/notifications_repository.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_local_notifications/flutter_local_notifications.dart';
+import 'package:flutter_timezone/flutter_timezone.dart';
+import 'package:timezone/data/latest_all.dart' as tz;
+import 'package:timezone/timezone.dart' as tz;
+
+final log = printer("NotificationsRepository");
+
+class NotificationsRepository implements INotificationsRepository {
+  late FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin;
+
+  NotificationsRepository({required this.flutterLocalNotificationsPlugin});
+
+  tz.TZDateTime nextInstanceOfTime(TimeOfDay time, tz.Location localTimeZOne) {
+    final tz.TZDateTime now = tz.TZDateTime.now(localTimeZOne);
+
+    tz.TZDateTime scheduledDate = tz.TZDateTime(
+        tz.local, now.year, now.month, now.day, time.hour, time.minute);
+    if (scheduledDate.isBefore(now)) {
+      scheduledDate = scheduledDate.add(const Duration(days: 1));
+    }
+
+    log.i("Scheduling alarm at $scheduledDate");
+    return scheduledDate;
+  }
+
+  @override
+  Future<void> zonedScheduleNotification(TimeOfDay time) async {
+    try {
+      final permissionsEnabled = await areNotificationsEnabled();
+
+      log.w("permissions enabled = $permissionsEnabled");
+
+      if (!permissionsEnabled) {
+        // We can request permission from within the App for >= Android 13
+        final arePermissionsGranted = await requestPermission();
+        if (!arePermissionsGranted) {
+          throw Exception("Notification permissions are not enabled");
+        }
+      }
+
+      // inititalize time zones
+      tz.initializeTimeZones();
+      final String? timeZoneName = await FlutterTimezone.getLocalTimezone();
+      tz.setLocalLocation(tz.getLocation(timeZoneName!));
+
+      log.i("Local timezone = ${tz.local}");
+
+      // cancel all previously scheduled notifications before scheduling new ones
+      cancelAllNotifications();
+
+      await flutterLocalNotificationsPlugin.zonedSchedule(
+          0,
+          'Time to Journal!',
```

Comment:
here it is hardcoded, as we don't have access to buildcontext

Reply:
@micedreams Had to migrate to `flutter_intls` package for localizing this change, have created separate PR for it: https://github.com/SankethBK/diaryvault/pull/97/files

Reply:
Next time switch the base branch from the master branch to the branch you are cascading from....changes made in the parent branch won't be there in the new PR and you will not have to deal with merge conflicts... GitHub will take care of it.